test/app-tap/datetime.test.lua

```diff
@@ -1830,6 +1830,565 @@ test:test("Parse strptime format", function(test)
+            buf = '09/09/04',
+            fmt = '%D',
+        },
+        -- %e
+        {
+            buf = ' 30',
+            fmt = '%e',
+        },
+        -- %E, modifier, skipped.
+        -- %F
+        {
+            buf = '1970-01-01T03:00:00.125',
+            fmt = '%FT%T.%f',
+        },
+        {
+            buf = '1970-01-01T03:00:00.125 MSK',
+            fmt = '%FT%T.%f %Z',
+        },
+        {
+            buf = '1970-01-01T03:00:00.125+0300',
+            fmt = '%FT%T.%f%z',
+        },
+        {
+            buf = '1970-01-01',
+            fmt = '%F',
+        },
+        -- %f, Tarantool-specific.
+        {
+            buf = '100',
+            fmt = '%f',
+        },
+        -- %G
+        {
+            buf = '2024',
+            fmt = '%G',
+        },
+        -- %g, it is not possible to roundtrip, skipped.
+        -- %h, same as %b
+        {
+            buf = 'Jan',
+            fmt = '%b',
+        },
+        {
+            buf = 'Feb',
+            fmt = '%b',
+        },
+        {
+            buf = 'Mar',
+            fmt = '%b',
+        },
+        {
+            buf = 'Apr',
+            fmt = '%b',
+        },
+        {
+            buf = 'May',
+            fmt = '%b',
+        },
+        {
+            buf = 'Jun',
+            fmt = '%b',
+        },
+        {
+            buf = 'Jul',
+            fmt = '%b',
+        },
+        {
+            buf = 'Aug',
+            fmt = '%b',
+        },
+        {
+            buf = 'Sep',
+            fmt = '%b',
+        },
+        {
+            buf = 'Oct',
+            fmt = '%b',
+        },
+        {
+            buf = 'Nov',
+            fmt = '%b',
+        },
+        {
+            buf = 'Dec',
+            fmt = '%b',
+        },
+        -- %H
+        {
+            buf = '00',
+            fmt = '%H',
+        },
+        {
+            buf = '23',
+            fmt = '%H',
+        },
+        -- %I
+        {
+            buf = '01',
+            fmt = '%I',
+        },
+        {
+            buf = '12',
+            fmt = '%I',
+        },
+        -- %j
+        {
+            buf = '001',
+            fmt = '%j',
+        },
+        {
+            buf = '31-12-2024 366',
+            fmt = '%d-%m-%Y %j',
+        },
+        -- %k
+        {
+            buf = ' 0',
+            fmt = '%k',
+        },
+        {
+            buf = '23',
+            fmt = '%k',
+        },
+        -- %l
+        {
+            buf = ' 1',
+            fmt = '%l',
+        },
+        {
+            buf = '12',
+            fmt = '%l',
+        },
+        -- %m
+        {
+            buf = '01',
+            fmt = '%m',
+        },
+        {
+            buf = '12',
+            fmt = '%m',
+        },
+        -- %M
+        {
+            buf = '00',
+            fmt = '%M',
+        },
+        {
+            buf = '59',
+            fmt = '%M',
+        },
+        -- %n
+        {
+            buf = '\n',
+            fmt = '%n',
```

Comment:
Maybe, it is better to add also a meaningful test with the full date separated by `%n`?
Feel free to ignore.

Reply:
Added:


```diff
--- a/test/app-tap/datetime.test.lua
+++ b/test/app-tap/datetime.test.lua
@@ -2276,6 +2276,11 @@ test:test("Parse with a custom format and format back to string", function(test)
             fmt = '%n',
             dt = date.new({year = 1970, month = 01, day = 01}),
         },
+        {
+            buf = '\n 2025-01-13 \n',
+            fmt = '%n %Y-%m-%d %n',
+            dt = date.new({year = 2025, month = 01, day = 13}),
+        },
         -- %O the same as %E*, skipped.
         -- %p is replaced by national representation of either
         -- "ante meridiem" (a.m.) or "post meridiem" (p.m.) as
```

Reply:
My bad, I mean `%Y%n%m%n%d`, feel free to ignore, though.

Reply:
Thanks!